modules/javafx.controls/src/main/java/javafx/scene/control/ChoiceBox.java

```diff
@@ -480,15 +479,25 @@ public String getName() {
 
     /**
      * Opens the list of choices.
+     *
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
      */
     public void show() {
-        if (!isDisabled()) setShowing(true);
+        Toolkit.getToolkit().checkFxUserThread();
+        if (!isDisabled()) {
+            setShowing(true);
+        }
     }
```

Comment:
Would it be a good idea to move the check `Toolkit.getToolkit().checkFxUserThread();` to a new method `show()` in Parent class `Control`? And may be similarly to Parent classes of other classes.